### PR TITLE
[release/v2.14] use Flatcar for base k8s conformance tests (#5930)

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
This is a manual backport of #5930.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Review by xmudrii — APPROVED** [review]
/lgtm

File: .prow.yaml [hunk 1]
```diff
@@ -376,7 +376,7 @@ presubmits:
           limits:
             memory: 4Gi
 
-  - name: pre-kubermatic-e2e-aws-coreos-1.18
+  - name: pre-kubermatic-e2e-aws-flatcar-1.18
```
Comment (kron4eg) [inline .prow.yaml]: "error":"error validating .prow.yaml: failed to validate .prow.yaml: duplicated presubmit job: pre-kubermatic-e2e-aws-flatcar-1.18"

**Review by xmudrii — APPROVED** [review]
/lgtm

**Review by xmudrii — APPROVED** [review]
/lgtm

**Pull request merged**